src/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -79,5 +78,211 @@ public static void ThrownFromReader()
                 Assert.Equal(8, inner.BytePositionInLine);
             }
         }
+
+        [Fact]
+        public static void PathForDictionaryFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<Dictionary<string, int>>(@"{""Key1"":1, ""Key2"":bad}");
+                Assert.True(false, "Expected JsonException was not thrown.");
+            }
+            catch (JsonException e)
+            {
+                Assert.Equal("$.Key2", e.Path);
+            }
+        }
+
+        [Fact]
+        public static void PathForArrayFails()
+        {
+            try
+            {
+                JsonSerializer.Parse<int[]>(@"[1, bad]");
```

Comment:
What happens if the JSON is `@"[1"`

Reply:
What happens if the JSON is @"[1 /* comment starts but doesn't end"

Reply:
The semantics are currently determined by the reader on what it treats as delimiters or separators for a given unit.

For `[1` it would be `$[0]`
For `[1 /* comment` it would be `$[1]`

Reply:
> For `[1 /* comment` it would be `$[1]`

`$[1]` isn't correct. There is no second item in `[1 /* comment`.

The index shouldn't be increased unless a comma is read.

Reply:
Situations like this is why I kept things simple in Json.NET and only included the known parent segments.

Reply:
Same response as above @ahsonkhan

Reply:
@steveharter, can you please file an issue to track this issue

Reply:
https://github.com/dotnet/corefx/issues/39478